dtfs-central-api/src/v1/controllers/portal/gef-facility/update-facility.controller.js

```diff
@@ -1,62 +1,81 @@
-        throw new Error('Invalid Deal Id');
-      }
+    if (dealId !== undefined && !ObjectId.isValid(dealId)) {
+      throw new InvalidDealIdError(dealId);
+    }
 
-      updatedFacility = await facilitiesCollection.findOneAndUpdate(
-        { _id: { $eq: ObjectId(id) } },
-        { $set: updateBody },
-        { returnNewDocument: true, returnDocument: 'after' },
-      );
+    const facilityUpdateWithAuditRecord = { ...facilityUpdate, auditRecord };
 
-      if (updatedFacility) {
-        // update facilitiesUpdated timestamp in the deal
-        const dealUpdateObj = { facilitiesUpdated: new Date().valueOf() };
+    const updatedFacility = await facilitiesCollection.findOneAndUpdate(
+      { _id: { $eq: ObjectId(facilityId) } },
+      { $set: facilityUpdateWithAuditRecord },
+      { returnNewDocument: true, returnDocument: 'after' },
+    );
 
-        await dealsCollection.updateOne({ _id: { $eq: ObjectId(dealId) } }, { $set: dealUpdateObj });
-      }
-    }
+    // update facilitiesUpdated timestamp in the deal
+    const dealUpdateObj = { facilitiesUpdated: new Date().valueOf(), auditRecord };
+
+    await dealsCollection.updateOne({ _id: { $eq: ObjectId(dealId) } }, { $set: dealUpdateObj });
 
     return updatedFacility;
   } catch (error) {
     console.error('Unable to update the facility %o', error);
-    return {};
+    throw error;
   }
 };
 exports.updateFacility = updateFacility;
 
 exports.updateFacilityPut = async (req, res) => {
-  if (ObjectId.isValid(req.params.id)) {
-    const facilityId = req.params.id;
-    const facilityUpdate = req.body;
+  const {
+    params: { id: facilityId },
+    body: { facilityUpdate, auditDetails },
+  } = req;
+
+  try {
+    if (!ObjectId.isValid(facilityId)) {
+      throw new InvalidFacilityIdError(facilityId);
+    }
 
-    const updatedFacility = await updateFacility(facilityId, facilityUpdate);
+    validateAuditDetails(auditDetails);
 
-    if (updatedFacility) {
-      return res.status(200).json(updatedFacility);
+    const updatedFacility = await updateFacility({ facilityId, facilityUpdate, auditDetails });
+    return res.status(200).json(updatedFacility);
+  } catch (error) {
+    if (error instanceof InvalidAuditDetailsError) {
+      return res.status(error.status).send({
+        status: error.status,
+        message: `Invalid auditDetails, ${error.message}`,
+      });
     }
 
-    return res.status(404).send({ status: 404, message: "The facility ID doesn't exist" });
-  }
+    if (error instanceof FacilityNotFoundError) {
+      return res.status(404).send({ status: 404, message: error.message });
+    }
 
-  return res.status(400).send({ status: 400, message: 'Invalid Facility Id' });
+    if (error instanceof InvalidDealIdError || error instanceof InvalidFacilityIdError) {
+      return res.status(400).send({ status: 400, message: error.message });
+    }
+    return res.status(500).send({ status: 500, error });
+  }
```

Comment:
Ideally could replace this with

```suggestion
    if (error instanceof ApiError) {
      return res.status(error.status).send({ status: error.status, message: error.message });
    }
```

Reply:
genius